src/main/kotlin/com/kllawesome/api/App.kt

```diff
@@ -0,0 +1,38 @@
+package com.kllawesome.api
+
+import com.kllawesome.api.util.coroutineHandler
+import io.vertx.core.http.HttpServer
+import io.vertx.ext.web.Router
+import io.vertx.ext.web.RoutingContext
+import io.vertx.ext.web.handler.BodyHandler
+import io.vertx.kotlin.coroutines.CoroutineVerticle
+import io.vertx.kotlin.coroutines.awaitResult
+
+
+class App : CoroutineVerticle() {
+
+    suspend override fun start() {
+        // Build Vert.x Web router
+        val router = Router.router(vertx)
+        router.route().handler(BodyHandler.create())
+        router.post("/api/upload").coroutineHandler { ctx -> uploadConfig(ctx) }
+
+        // Start the server
+        awaitResult<HttpServer> {
+            vertx.createHttpServer()
+                    .requestHandler(router::accept)
+                    .listen(config.getInteger("http.port", 8080), it)
+        }
+    }
+
+    suspend fun uploadConfig(ctx: RoutingContext) {
+        val jsonBody = ctx.bodyAsJson
+        val configName = jsonBody.getString("configTitle")
+        val config = jsonBody.getString("config")
```

Comment:
For config info the frontend is currently going to submits `githubUrl`, `category`, and `configName` (or Title depending on what we decide.